app/backend/src/tests/test_admin.py

```diff
@@ -645,4 +645,34 @@ def test_ListUserIds(db):
         assert res.user_ids == []
 
 
+def test_EditReferenceText(db):
+    super_user, super_token = generate_user(is_superuser=True)
+    test_new_text = "New Text"
+
+    user1, user1_token = generate_user()
+    user2, user2_token = generate_user()
+
+    with session_scope() as session:
+        reference = Reference(
+            from_user_id=user1.id,
+            to_user_id=user2.id,
+            reference_type=ReferenceType.friend,
+            text="Old Text",
+            rating=0.2,
+            was_appropriate=True,
+        )
+        session.add(reference)
+        session.commit()
+
+        with real_admin_session(super_token) as api:
+            res = api.EditReferenceText(
+                admin_pb2.EditReferenceTextReq(reference_id=reference.id, new_text=test_new_text)
+            )
+
+        session.expire_all()
+
+        modified_reference = session.execute(select(Reference).where(Reference.id == reference.id)).scalar_one_or_none()
+        assert modified_reference.text == test_new_text
+
+
```

Comment:
This test is good. I guess in general it's slightly better to do the testing via the API (make ref via API, edit like here via API, then list/get via API). That way you're testing it all. But this is good too, no need to change.